### PR TITLE
Modify rest api post schema compatibility strategy

### LINKED ISSUE #4821: Pulsar schema api should respect to namespace level compatibility
**Describe the bug**
Currently the schema restful api is using `SchemaCompatibilityStrategy.FULL`

https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/SchemasResource.java#L268

It should use the compatibility strategy configured at namespace level.

### PR DESCRIPTION
**Pull request by congbobo184**
### Motivation
To fix #4821 

Does this pull request potentially affect one of the following parts:
If yes was chosen, please highlight the changes

Dependencies (does it add or upgrade a dependency): (no)
The public API: (no)
The schema: (yes)
The default values of configurations: (no)
The wire protocol: (no)
The rest endpoints: (no)
The admin cli options: (no)
Anything that affects deployment: (no)

### Documentation
Does this pull request introduce a new feature? (yes / no)
If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
If a feature is not applicable for documentation, explain why?
If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by congbobo184** [comment]
run cpp tests

**Comment by congbobo184** [comment]
run java8 tests

**Comment by congbobo184** [comment]
run java8 tests

**Comment by congbobo184** [comment]
run Integration Tests

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/SchemasResource.java [hunk 1]
```diff
@@ -310,31 +310,49 @@ public void postSchema(
...
+                            Response.serverError().build()
+                    );
+                }
+                return null;
+            });
+        }).exceptionally(error -> {
+            if (error.getCause() instanceof RestException) {
                 response.resume(Response.status(
-                    422, /* Unprocessable Entity */
+                    404, /* Unprocessable Entity */
```
Comment (sijie) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/SchemasResource.java]: if it is RestException, you need to return the status of the RestException, no?

Comment (congbobo184) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/SchemasResource.java]: OK, I will change it then return the response

**Comment by congbobo184** [comment]
run java8 tests

**Pull request merged**